### PR TITLE
refactor downloadMultiple media endpoint

### PR DESCRIPTION
**Pull request by SirSplasch**
add AddressMediaScope global scope

### CODE REVIEW

File: src/Http/Controllers/MediaController.php [hunk 1]
```diff
@@ -76,17 +77,18 @@ public function deleteCollection(Request $request, MediaService $mediaService):
...
-        $fileName = Str::finish($request->input('filename') ?? 'media', '.zip');
-        $ids = is_array($ids) ? $ids : explode(',', $ids);
+        $data = $request->validated();
 
-        if (! $ids) {
-            abort(422);
-        }
+        $fileName = Str::finish(data_get($data, 'filename') ?: 'media', '.zip');
+        $ids = is_array(data_get($data, 'ids')) ?
+            data_get($data, 'ids') : explode(',', data_get($data, 'ids'));
```
Comment (patrickweh) [inline src/Http/Controllers/MediaController.php:86]: Break line

**Review by patrickweh — CHANGES_REQUESTED** [review]

File: src/Models/Scopes/AddressMediaScope.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            return;
+        }
+
+        // Only Apply Scope if User is an Address
+        if (! Auth::user() instanceof Address) {
+            return;
+        }
+
+        $builder->where(function (Builder $query) {
+            $query->where('model_type', '!=', morph_alias(Order::class))
```
Comment (patrickweh) [inline src/Models/Scopes/AddressMediaScope.php:27]: whereNot, looks better...

**Review by patrickweh — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/MediaController.php [hunk 3]
```diff
@@ -76,17 +77,19 @@ public function deleteCollection(Request $request, MediaService $mediaService):
...
-        $ids = $request->input('ids');
-        $fileName = Str::finish($request->input('filename') ?? 'media', '.zip');
-        $ids = is_array($ids) ? $ids : explode(',', $ids);
+        $data = $request->validated();
 
-        if (! $ids) {
-            abort(422);
-        }
+        $fileName = Str::finish(data_get($data, 'filename') ?: 'media', '.zip');
+        $ids = is_array(data_get($data, 'ids')) ?
```
Comment (patrickweh) [inline src/Http/Controllers/MediaController.php:85]: Usually we begin with the operator in the line...

**Review by patrickweh — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/MediaController.php (hunk 3 above)
Comment (SirSplasch) [reply src/Http/Controllers/MediaController.php:85]: won't do. To me this looks cleaner.

**Review by patrickweh — APPROVED** [review]

**Pull request merged**